tx-pool/src/process.rs

```diff
@@ -652,117 +654,182 @@ impl TxPoolService {
 
         let data_loader = snapshot.as_data_loader();
 
-        let completed = if let Some(ref entry) = cached {
-            match entry {
-                CacheEntry::Completed(completed) => {
-                    let ret = TimeRelativeTransactionVerifier::new(
-                        Arc::clone(&rtx),
-                        Arc::clone(&self.consensus),
-                        data_loader,
-                        tx_env,
-                    )
-                    .verify()
-                    .map_err(Reject::Verification);
-                    try_or_return_with_snapshot!(ret, snapshot);
-                    *completed
-                }
-                CacheEntry::Suspended(_) => {
-                    return Some((Ok(ProcessResult::Suspended), snapshot));
-                }
-            }
-        } else {
-            let is_chunk_full = self.is_chunk_full().await;
-
-            let ret = block_in_place(|| {
-                let verifier = ContextualTransactionVerifier::new(
+        let completed = match cached {
+            Some(completed) => {
+                let ret = TimeRelativeTransactionVerifier::new(
                     Arc::clone(&rtx),
                     Arc::clone(&self.consensus),
                     data_loader,
                     tx_env,
-                );
-
-                let (ret, fee) = verifier
-                    .resumable_verify(limit_cycles)
-                    .map_err(Reject::Verification)?;
-
-                match ret {
-                    ScriptVerifyResult::Completed(cycles) => {
-                        if let Err(e) = DaoScriptSizeVerifier::new(
+                )
+                .verify()
+                .map_err(Reject::Verification);
+                try_or_return_with_snapshot!(ret, snapshot);
```

Comment:
How about remove `try_or_return_with_snapshot!` macro, and use:
```rust
        let state:VerifyResult = ret.map_err(|err| snapshot)?;
```

Reply:
`_resumeble_process_tx` is not returning a `Result`.